src/order/filter/interval.lean

```diff
@@ -0,0 +1,132 @@
+/-
+Copyright (c) 2020 Yury G. Kudryashov. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Author: Yury G. Kudryashov
+-/
+import data.set.intervals.ord_connected
+import order.filter.lift
+import order.filter.at_top_bot
+
+/-!
+# Interval-generated filters
+
+We say that a filter `f` on a `preorder` is interval generated if each set `s ∈ f` includes
+an `ord_connected` subset `t ∈ f`. Equivalently, `f` has a basis `{s i | p i}` which consists
+of `ord_connected` sets.
+-/
+
+variables {α β : Type*}
+
+open_locale classical filter
+
+open set function
+
+namespace filter
+
+variables [preorder α]
+
+/--
+A filter `f` on a `preorder` is interval generated if each set `s ∈ f` includes an `ord_connected`
+subset `t ∈ f`.
+-/
+class is_interval_generated (f : filter α) : Prop :=
```

Comment:
Would it be more consistent and efficient to also make `ord_connected` a class?

Reply:
It has conversions to/from `convex` and `is_preconnected`. Another difficulty is that some lemmas (e.g., `ord_connected_sUnion`) have non-instance arguments. Possibly I can make it a `@[class] def` but I'd prefer not to do it right now.